src/main/groovy/org/shipkit/internal/gradle/plugin/PluginUtil.java

```diff
@@ -0,0 +1,27 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.util.PatternSet;
+
+import java.io.File;
+import java.util.Set;
+
+public class PluginUtil {
+    static final String DOT_PROPERTIES = ".properties";
+
+    static Set<File> discoverGradlePluginPropertyFiles(Project project) {
+        final JavaPluginConvention java = project.getConvention().getPlugin(JavaPluginConvention.class);
+        FileTree resources = java.getSourceSets().getByName("main").getResources();
+        FileTree plugins = resources.matching(new PatternSet().include("META-INF/gradle-plugins/*" + DOT_PROPERTIES));
+        return plugins.getFiles();
+    }
+
+    static Set<File> discoverGradlePlugins(Project project) {
+        final JavaPluginConvention java = project.getConvention().getPlugin(JavaPluginConvention.class);
+        FileTree resources = java.getSourceSets().getByName("main").getAllJava();
+        FileTree plugins = resources.matching(new PatternSet().include("**/*Plugin.java").include("**/*Plugin.groovy"));
```

Comment:
seems to work, see the most recent comment showing missing properties files in shipkit.